ruby/ext/google/protobuf_c/message.c

```diff
@@ -329,6 +329,30 @@ VALUE Message_inspect(VALUE _self) {
   return str;
 }
 
+
+VALUE Message_to_h(VALUE _self) {
+  MessageHeader* self;
+  TypedData_Get_Struct(_self, MessageHeader, &Message_type, self);
+
+  VALUE hash = rb_hash_new();
+  rb_gc_register_address(&hash);
```

Comment:
ha!  yes; sorry, I should have dived into the docs a bit more on that one.  thanks for the explanation.